Sources/TuistServer/Cache/CacheStorageFactorying.swift

```diff
@@ -7,4 +7,6 @@ import XcodeGraph
 @Mockable
 public protocol CacheStorageFactorying {
     func cacheStorage(config: Config) async throws -> CacheStoring
+    /// - Returns: Cache storage that only works with the local cache.
+    func cacheLocalStorage() async throws -> CacheStoring
```

Comment:
I did consider refactoring `CacheStorage` to have its own protocol, so you could run `store(noUpload: true)` and `CacheStorage` would be responsible for determining whether `cacheRemoteStorage.store` should be called or not.

I felt in the end this was a bit cleaner as we keep a single `CacheStoring` protocol, but I don't feel strongly about this one.